src/client.js

```diff
@@ -221,6 +221,21 @@ Client.prototype.connect = function(args) {
 	});
 };
 
+Client.prototype.setPassword = function(hash) {
+	var client = this;
+	client.manager.updateUser(client.name, {password:hash});
+	// re-read the hash off disk to ensure we use whatever is saved. this will
+	// prevent situations where the password failed to save properly and so
+	// a restart of the server would forget the change and use the old
+	// password again.
+	var user = client.manager.readUserConfig(client.name);
```

Comment:
I'm not sure I understand that line and comment: if the password was not saved properly (so, not saved), why do we need to re-read the config, since the password will be unchanged? Also, doesn't restarting the server trigger a new config file read?